tests/sasl_plain_ut/sasl_plain_ut.c

```diff
@@ -86,95 +115,94 @@ TEST_FUNCTION(saslplain_create_with_valid_args_succeeds)
 {
 	// arrange
-	amqp_frame_codec_mocks mocks;
 	SASL_PLAIN_CONFIG sasl_plain_config = { "test_authcid", "test_pwd", "test_authzid" };
 
-	EXPECTED_CALL(mocks, amqpalloc_malloc(IGNORED_NUM_ARG))
-		.SetReturn((void*)NULL);
+	EXPECTED_CALL(gballoc_malloc(IGNORED_NUM_ARG))
+		.SetReturn(NULL);
 
 	// act
 	CONCRETE_SASL_MECHANISM_HANDLE result = saslplain_create(&sasl_plain_config);
 
 	// assert
-	ASSERT_IS_NULL(result);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+    ASSERT_IS_NULL(result);
 }
 
 /* Tests_SRS_SASL_PLAIN_01_002: [If allocating the memory needed for the saslplain instance fails then saslplain_create shall return NULL.] */
 TEST_FUNCTION(when_allocating_memory_for_the_config_fails_then_saslplain_create_fails)
 {
 	// arrange
-	amqp_frame_codec_mocks mocks;
 	SASL_PLAIN_CONFIG sasl_plain_config = { "test_authcid", "test_pwd", "test_authzid" };
 
-	EXPECTED_CALL(mocks, amqpalloc_malloc(IGNORED_NUM_ARG));
-	EXPECTED_CALL(mocks, amqpalloc_malloc(IGNORED_NUM_ARG))
-		.SetReturn((void*)NULL);
-	EXPECTED_CALL(mocks, amqpalloc_free(IGNORED_PTR_ARG));
+	EXPECTED_CALL(gballoc_malloc(IGNORED_NUM_ARG));
+	EXPECTED_CALL(gballoc_malloc(IGNORED_NUM_ARG))
+		.SetReturn(NULL);
+	EXPECTED_CALL(gballoc_free(IGNORED_PTR_ARG));
 
 	// act
 	CONCRETE_SASL_MECHANISM_HANDLE result = saslplain_create(&sasl_plain_config);
 
 	// assert
-	ASSERT_IS_NULL(result);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+    ASSERT_IS_NULL(result);
 }
 
 /* Tests_SRS_SASL_PLAIN_01_003: [If the config argument is NULL, then saslplain_create shall fail and return NULL.] */
 TEST_FUNCTION(saslplain_create_with_NULL_config_fails)
 {
 	// arrange
-	amqp_frame_codec_mocks mocks;
 
 	// act
 	CONCRETE_SASL_MECHANISM_HANDLE result = saslplain_create(NULL);
 
 	// assert
-	ASSERT_IS_NULL(result);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+    ASSERT_IS_NULL(result);
 }
 
 /* Tests_SRS_SASL_PLAIN_01_004: [If either the authcid or passwd member of the config structure is NULL, then saslplain_create shall fail and return NULL.] */
 TEST_FUNCTION(saslplain_create_with_NULL_authcid_fails)
 {
 	// arrange
-	amqp_frame_codec_mocks mocks;
 	SASL_PLAIN_CONFIG sasl_plain_config = { NULL, "test_pwd", "test_authzid" };
 
 	// act
 	CONCRETE_SASL_MECHANISM_HANDLE result = saslplain_create(&sasl_plain_config);
 
 	// assert
-	ASSERT_IS_NULL(result);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+    ASSERT_IS_NULL(result);
 }
 
 /* Tests_SRS_SASL_PLAIN_01_004: [If either the authcid or passwd member of the config structure is NULL, then saslplain_create shall fail and return NULL.] */
 TEST_FUNCTION(saslplain_create_with_NULL_passwd_fails)
 {
 	// arrange
-	amqp_frame_codec_mocks mocks;
 	SASL_PLAIN_CONFIG sasl_plain_config = { "test_authcid", NULL, "test_authzid" };
 
```

Comment:
Not needed as it is done in test function init.

---
In reply to: [104450017](https://github.com/Azure/azure-uamqp-c/pull/35#discussion_r104450017) [](ancestors = 104450017)